js/renovation/ui/grids/data_grid_next/types.ts

```diff
@@ -25,14 +26,33 @@ export interface RowTemplateProps {
   rowIndex: number;
 }
 
-export interface ColumnInternal {
+export type ColumnDataType = 'string' | 'number' | 'date' | 'boolean' | 'object' | 'datetime';
+export type ColumnAlignment = 'left' | 'right' | 'center';
+
+export interface Column {
   dataField?: string;
 
-  headerCssClass?: string;
+  caption?: string;
+
+  alignment?: ColumnAlignment;
+
+  format?: Format;
+
+  cssClass?: string;
+
+  width?: number | string;
+
+  dataType?: ColumnDataType;
+
+  calculateCellValue?: (data: RowData) => unknown;
 
   cellTemplate?: JSXTemplate<{ data: RowData }, 'data'>;
 
-  headerTemplate?: JSXTemplate;
+  headerCellTemplate?: JSXTemplate;
+}
+
+export interface ColumnInternal extends Column {
+  headerCssClass?: string;
```

Comment:
`dataType`, `calculateCellValue` and others are not optional in `ColumnInternal`